src/django_components/finders.py

```diff
@@ -0,0 +1,144 @@
+import os
+from fnmatch import fnmatch
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+
+from django.contrib.staticfiles.apps import StaticFilesConfig
+from django.contrib.staticfiles.finders import BaseFinder
+from django.contrib.staticfiles.utils import get_files
+from django.core.checks import CheckMessage, Error, Warning
+from django.core.files.storage import FileSystemStorage
+from django.utils._os import safe_join
+
+from django_components.app_settings import app_settings
+from django_components.template_loader import get_dirs
+
+# To keep track on which directories the finder has searched the static files.
+searched_locations = []
+
+
+# Custom Finder for staticfiles that searches for all files within the directories
+# defined by `COMPONENTS.dirs`.
+#
+# This is what makes it possible to define JS and CSS files in the directories as
+# defined by `COMPONENTS.dirs`, but still use the JS / CSS files with `static()` or
+# `collectstatic` command.
+class ComponentsFileSystemFinder(BaseFinder):
+    """
+    A static files finder based on ``FileSystemFinder``.
+
+    Differences:
+    - This finder uses ``COMPONENTS.dirs`` setting to locate files instead of ``STATICFILES_DIRS``.
+    - This finder ignores Python and HTML files (``.py``, ``.pyc``, ``.html``, ``.django``)
+    - If ``COMPONENTS.dirs`` is not set or empty, defaults to ``settings.BASE_DIR / "components"``
+    """
+
+    IGNORE_PATTERNS = StaticFilesConfig.ignore_patterns + ["*.html", "*.django", "*.py", "*.pyc"]
+
+    def __init__(self, app_names: Any = None, *args: Any, **kwargs: Any) -> None:
+        component_dirs = [str(p) for p in get_dirs()]
```

Comment:
It's still the Loader class where we define how we search the filesystem for component files. Here, inside the Finder, we use `get_dirs()`, which calls / uses the loader.